_posts/2025-01-17-riva128-part-1.md

```diff
@@ -0,0 +1,305 @@
+
+**`0x06` (Pattern)**: Pattern used for bitblit and other blits 
+
+**`0x07` (Rectangle)**: Up to 16 rectangles with size and position represented as a 32-bit value (Y as high 16 bits, X as low 16)
+
+**`0x08` (Point)**: 
+
+An arbitrary point on the screen. Depending on the methods used to submit the object, this object can take the form of:
+
+* Up to 32 points, each with a single arbitrary 32-bit colour (probably BGRA format) and 16-bit size and position values.
+* Up to 16 points, each  with a single arbitrary 32-bit colour (probably BGRA format) and 32-bit size and position values.
+* Up to 16 points, making up a polygon, with an arbitrary 32-bit colour for each polygon line (probably BGRA format) and 16-bit size and position values.
+
+**`0x09` (Line)**:
+
+An arbitrary line on the screen. Depending on the methods used to submit the object, this object can take the form of:
+
+* Up to 16 lines, each with a single arbitrary 32-bit colour (probably BGRA format) and 16-bit size and position values.
+* Up to 8 lines, each with a single arbitrary 32-bit colour (probably BGRA format) and 32-bit size and position values.
+* Up to 32 lines, each making up a polygon, with a single arbitrary 32-bit colour (probably BGRA format) and 16-bit size and position values.
+* Up to 16 lines, each making up a polygon, with a single arbitrary 32-bit colour (probably BGRA format) and 32-bit size and position values. 
+* Up to 16 lines, each making up a polygon, with an arbitrary 32-bit colour for each polygon line (probably BGRA format) and 16-bit size and position values.
+
+**`0x0A` (Lin)**: The exact same as the Line object, but the starting and ending pixels are not drawn for each line.
+
+**`0x0B` (Triangle)**: 
+
+A basic (presumably pre-transformed...?) 2D triangle. Depending on the methods used to submit the object, this object can take the form of:
+
+* A single triangle with a single arbitrary 32-bit colour and three 16-bit position values for each of the triangle's vertexes.
+* A single triangle with a single arbitrary 32-bit colour for the entire mesh, and three 16-bit position values for each of the triangle's vertexes.
+* A part of a mesh of up to 32 triangles with a single arbitrary 32-bit colour and two 16-bit position values for each of the points on the mesh.
+* A part of a mesh of up to 16 triangles with a single arbitrary 32-bit colour and two 32-bit position values for each of the points on the mesh.
+* A set of up to 8 triangles with a single arbitrary 32-bit colour for the entire mesh, and three 16-bit position values for each of the triangle's vertexes.
+* A part of a mesh of up to 16 triangles with a 32-bit colour and two 32-bit position values for each of the points on the mesh.
+
+**`0x0C` (Windows 95 GDI Text Acceleration)**: A piece of hardware functionality intended to acce. This is a very complicated set of clipping logic that won't be covered until Part 3 - it's too long for this part, and I don't fully understand it yet.
```

Comment:
acce?

Reply:
i managed to only write half a sentence. Fixed